esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -0,0 +1,168 @@
+
+        Fix dtype value of coordinates and coordinate bounds.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            if cube.coord('time').units.calendar == 'standard':
+                cube.coord('time').units = Unit(
+                    'days since 1850-1-1 00:00:00',
+                    calendar='proleptic_gregorian'
+                )
+            for coord in cube.coords():
+                if coord.dtype in ['>f8', '>f4']:
+                    coord.points = coord.core_points().astype(
+                        np.float64, casting='same_kind')
+                    if coord.bounds is not None:
+                        coord.bounds = coord.core_bounds().astype(
+                            np.float64, casting='same_kind')
+        return cubes
+
+
+class AllVars(Fix):
+    """General CORDEX grid fix."""
+
+    def _fix_rotated_coords(self, cube):
+        """Fix rotated coordinates."""
+        data_domain = self.extra_facets['domain']
+        domain = cx.cordex_domain(data_domain, add_vertices=True)
+        domain_info = cx.domain_info(data_domain)
+        for dim_coord in ['rlat', 'rlon']:
+            old_coord = cube.coord(domain[dim_coord].standard_name)
+            old_coord_dims = old_coord.cube_dims(cube)
+            points = domain[dim_coord].data
+            coord_system = iris.coord_systems.RotatedGeogCS(
+                grid_north_pole_latitude=domain_info['pollat'],
+                grid_north_pole_longitude=domain_info['pollon']
+            )
+            new_coord = iris.coords.DimCoord(
+                points,
+                var_name=dim_coord,
+                standard_name=domain[dim_coord].standard_name,
+                long_name=domain[dim_coord].long_name,
+                units=Unit('degrees'),
+                coord_system=coord_system,
+            )
+            new_coord.guess_bounds()
+            cube.remove_coord(old_coord)
+            cube.add_dim_coord(new_coord, old_coord_dims)
+
+    def _fix_geographical_coords(self, cube):
+        """Fix geographical coordinates."""
+        data_domain = self.extra_facets['domain']
+        domain = cx.cordex_domain(data_domain, add_vertices=True)
+        for aux_coord in ['lat', 'lon']:
+            old_coord = cube.coord(domain[aux_coord].standard_name)
+            cube.remove_coord(old_coord)
+            points = domain[aux_coord].data
+            bounds = domain[f'{aux_coord}_vertices'].data
+            new_coord = iris.coords.AuxCoord(
+                points,
+                var_name=aux_coord,
+                standard_name=domain[aux_coord].standard_name,
+                long_name=domain[aux_coord].long_name,
+                units=Unit(domain[aux_coord].units),
+                bounds=bounds
+            )
+            cube.add_aux_coord(new_coord, (1, 2))
```

Comment:
This will fail for fx variables such as areacella, since the axis would be 0 and 1.

Reply:
Done as well, I generalised the dimension values